source/iopipe/json/serialize.d

```diff
@@ -845,118 +836,511 @@ unittest
     assert(cArr.length == 3);
 }
 
-void serializeImpl(T, Char)(scope void delegate(const(Char)[]) w, ref T val) if (__traits(isStaticArray, T))
+//alias OutputDelegate(Char) = void delegate(const(Char)[]);
+
+struct FormattingOptions
+{
+    bool lineBreaks;
+    bool spaceAfterColon;
+    private bool _spaceBetweenItems;
+    // Only makes sense if lineBreaks == false
+    bool spaceBetweenItems() => !lineBreaks && _spaceBetweenItems;
+    bool spaceBetweenItems(bool value)
+    {
+        _spaceBetweenItems = value;
+        return spaceBetweenItems;
+    }
+    bool trailingCommas;
+    bool preferUnquotedKeys;
+    bool preferSingleQuotes;
+    string quoteText() => preferSingleQuotes ? "'" : "\"";
+    private bool _indentation;
+    // Only makes sense if lineBreaks == true
+    bool indentation() => lineBreaks && _indentation;
+    bool indentation(bool value)
+    {
+        _indentation = value;
+        return indentation;
+    }
+    int indentationLevel;
+    string indentationText;
+    
+    FormattingOptions indent()
+    {
+        FormattingOptions fo = this;
+        fo.indentationLevel++;
```

Comment:
This should only happen if `indentationText` is not 0-length. Otherwise, you waste time calling `w("")`